UbuntuDrivers/kerneldetection.py

```diff
@@ -173,19 +185,25 @@ def get_linux_version(self):
         linux_image_meta = self.get_linux_image_metapackage()
         linux_version = ''
         try:
-            dependencies = self.apt_cache[linux_image_meta].candidate.\
-                             record['Depends']
+            # dependencies = self.apt_cache[linux_image_meta].candidate.\
+            #                  record['Depends']
+            records = apt_pkg.PackageRecords(self.apt_cache)
+            candidate = self.apt_depcache.get_candidate_ver(self.apt_cache[linux_image_meta])
+            records.lookup(candidate.file_list[0])
```

Comment:
You don't seem to be using the records, get rid of the initialization and the lookup